Lib/test/test_named_expressions.py

```diff
@@ -271,6 +271,12 @@ def test_named_expression_assignment_16(self):
         fib = {(c := a): (a := b) + (b := a + c) - b for __ in range(6)}
         self.assertEqual(fib, {1: 2, 2: 3, 3: 5, 5: 8, 8: 13, 13: 21})
 
+    def test_named_expression_assignment_17(self):
+        a = [1]
+        element = a[b:=0]
+        self.assertEqual(b, 0)
+        self.assertEqual(element, a[0])
+
```

Comment:
Can you add a test also with more complex indexing like [b:=0, c:=1]?

Reply:
We can only test that this does not raise a `SyntaxError`, right? Is there a builtin data structure where we can use indexing with a tuple?

Reply:
> We can only test that this does not raise a `SyntaxError`, right?

Yup. You can either use a class with custom `__getitem__` or a mock

Reply:
Done.